src/core/vdom/patch.js

```diff
@@ -237,7 +237,9 @@ export function createPatchFunction (backend) {
   function insert (parent, elm, ref) {
     if (isDef(parent)) {
       if (isDef(ref)) {
-        nodeOps.insertBefore(parent, elm, ref)
+        if (ref.parentNode && ref.parentNode === parent) {
```

Comment:
I think just `ref.parentNode === parent` is enough. `isDef(parent)` has already been checked.